biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/ChemCompGroupFactory.java

```diff
@@ -79,6 +77,8 @@ public static ChemComp getChemComp(String recordName){
 	public static void setChemCompProvider(ChemCompProvider provider) {
 		logger.debug("Setting new chem comp provider to "+provider.getClass().getCanonicalName());
 		chemCompProvider = provider;
+		// clear cache
+		cache.clear();
```

Comment:
Yes this looks familiar... I don't really remember what was the problem exactly but I'm pretty sure we've had a lot of subtle issues around how the cache is initialised and cleared. See for instance #313 

Let's anyway merge it so that we can test it for a while before the release. I know that @abradle is doing a lot of parallel parsing of files and he would be able to find problems with this.